security-analysis/security-analysis-api/src/test/java/com/powsybl/security/LimitViolationDetectionTest.java

```diff
@@ -78,4 +95,21 @@ protected void checkVoltage(Bus b, int voltageValue, Consumer<LimitViolation> co
     protected void checkVoltageAngle(VoltageAngleLimit voltageAngleLimit, double voltageAngleDifference, Consumer<LimitViolation> consumer) {
         LimitViolationDetection.checkVoltageAngle(voltageAngleLimit, voltageAngleDifference, consumer);
     }
+
+    @Test
+    void testLimitsComputer() {
+        Network network = EurostagTutorialExample1Factory.createWithFixedCurrentLimits();
+        List<LimitReduction> limitReductionList = new ArrayList<>();
+        LimitReduction reduction1 = LimitReduction.builder(LimitType.CURRENT, 0.5)
+            .withMonitoringOnly(false)
+            .withContingencyContext(ContingencyContext.none())
+            .withNetworkElementCriteria(new NetworkElementIdListCriterion(Set.of("NHV1_NHV2_1")))
+            .withLimitDurationCriteria(new PermanentDurationCriterion())
+            .build();
+        limitReductionList.add(reduction1);
+        DefaultLimitReductionsApplier computer = new DefaultLimitReductionsApplier(limitReductionList);
+        checkCurrent(network.getLine("NHV1_NHV2_1"), TwoSides.ONE, 1101, violationsCollector::add, computer);
+        Assertions.assertEquals(1, violationsCollector.size());
+        Assertions.assertEquals(0.5, violationsCollector.get(0).getLimitReduction());
```

Comment:
You could add more assertions:

```suggestion
        DefaultLimitReductionsApplier computer = new DefaultLimitReductionsApplier(limitReductionList);
        checkCurrent(network.getLine("NHV1_NHV2_1"), TwoSides.ONE, 251, violationsCollector::add, computer);
        Assertions.assertEquals(1, violationsCollector.size());
        Assertions.assertEquals(0.5, violationsCollector.get(0).getLimitReduction());
        Assertions.assertEquals(Integer.MAX_VALUE, violationsCollector.get(0).getAcceptableDuration());
        Assertions.assertEquals(251., violationsCollector.get(0).getValue(), 0.01);
        Assertions.assertEquals(500., violationsCollector.get(0).getLimit(), 0.01);
        Assertions.assertEquals(ThreeSides.ONE, violationsCollector.get(0).getSide());
```